module/Frontpage/src/Frontpage/Service/Poll.php

```diff
@@ -25,10 +25,19 @@ public function getNewestPoll()
      * Retrieves a poll by its id
      * @param int $pollId the id of the poll to retrieve
      * @return PollModel|null
+     *
+     * @throws \User\Permissions\NotAllowedException if the user isn't allowed to see unapproved polls
      */
     public function getPoll($pollId)
     {
-        return $this->getPollMapper()->findPollById($pollId);
+        $poll = $this->getPollMapper()->findPollById($pollId);
+        if ((null === $poll->getApprover()) && !$this->isAllowed('view_unapproved')) {
```

Comment:
I know for a fact that `is_null` isn't used everywhere else in the code (since I have always been using `null ===`). I usually prefer language constructions over functions. Neither am I sure that it is good to use `is_null`.

And the fact that it's a Yoda condition kinda is an old habit that dies hard (I think it was a code standard in ZF1 if you used a comparison with `null`).